src/lib/redux/inventoryData.ts

```diff
@@ -136,3 +138,29 @@ export const setCurrentInventoryIdInRedux = (inventoryId: string): void => {
 export const addProductToRedux = (product: ProductRecord): void => {
   store.dispatch(addProduct(product));
 };
+
+export const selectAmountPurchaseRequestedApproved = createSelector(
+    selectAllCurrentSitePurchaseRequestsArray,
```

Comment:
Make sure that you calculate the amount approvedfor just the current period not all periods